### PR TITLE
Support VS telemetry and NFW in service hub

### PR DESCRIPTION
**Pull request by heejaechang**
this should let us to have better insight on our service hub process

### CODE REVIEW

**Comment by heejaechang** [comment]
@srivatsn @mavasani @panopticoncentral @kirankulkarni747 can you take a look?

**Comment by heejaechang** [comment]
tagging @KevinH-MS and @rynowak 

what this means is that, now you guys can use our Logger to send VS telemetry in our service hub process if you want to. and also use FatalError or WatsonReporter to report NFW if you want to.

**Comment by rynowak** [comment]
That's pretty cool, is this through your `ILogger`? I don't think we have access to that.

**Comment by heejaechang** [comment]
@rynowak if you guys don't have access to it, probably need to make it available through your stub in our side. or you can just make TelemetrySession available to you from VSTelemetryLogger.Session, and can use VS Telemetry API directly if you want to (MS.VS.Telemetry nuget)

File: src/Workspaces/Remote/ServiceHub/Telemetry/VSTelemetryCache.cs [hunk 1]
```diff
@@ -0,0 +1,67 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using Microsoft.CodeAnalysis.Internal.Log;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Remote.Telemetry
```
Comment (heejaechang) [inline src/Workspaces/Remote/ServiceHub/Telemetry/VSTelemetryCache.cs]: this is basically port of one in VS layer to suite for service hub

File: src/Workspaces/Remote/ServiceHub/Telemetry/VSTelemetryLogger.cs [hunk 2]
```diff
@@ -0,0 +1,149 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading;
+using Microsoft.CodeAnalysis.Internal.Log;
+using Microsoft.VisualStudio.Telemetry;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Remote.Telemetry
+{
+    internal class VSTelemetryLogger : ILogger
```
Comment (heejaechang) [inline src/Workspaces/Remote/ServiceHub/Telemetry/VSTelemetryLogger.cs]: this is basically port of one in vs layer to suite better on service hub.

**Comment by heejaechang** [comment]
ping? can I get some code review?

**Comment by heejaechang** [comment]
@tmeschter now we can get NFW watson from our OOP process as well. just FYI. one from devenv will have devenv as exename and one from OOP will have roslyncodeanalysisservice

**Comment by tmeschter** [comment]
@heejaechang Do you mean ServiceHub.RoslynCodeAnalysisService32.exe?

**Comment by heejaechang** [comment]
@tmeschter yes. or you can do exeName contains "RoslynCodeAnalysisService32"

**Pull request merged**